pkg/pillar/containerd/oci.go

```diff
@@ -224,28 +223,49 @@ func (s *ociSpec) updateFromImageConfig(config v1.ImageConfig) error {
 }
 
 // UpdateMounts
-func (s *ociSpec) UpdateMounts(disks []types.DiskConfig) {
-	for i := range disks {
-		// Skipping root container disk
-		if i == 0 || isFile(disks[i].FileLocation) {
+func (s *ociSpec) UpdateMounts(disks []types.DiskStatus, root string) {
+	s.Mounts = append(s.Mounts, specs.Mount{Type: "tmpfs", Source: "tmpfs", Destination: "/mnt"})
+	for id, disk := range disks {
+		// we may need additional filtering here, but for now assume that
+		// we can bind mount anything aside from FmtUnknown
+		srcPostfix := ""
+		switch disk.Format {
+		case zconfig.Format_FmtUnknown:
 			continue
+		case zconfig.Format_CONTAINER:
+			if path.Join(disk.MountDir) != "/" {
+				srcPostfix = "rootfs"
+			}
 		}
-		mount := specs.Mount{}
-		access := ""
-		if disks[i].ReadOnly {
-			access = "rbind:ro"
+
+		opts := []string{"rbind"}
+		if disk.ReadOnly {
+			opts = append(opts, "ro")
 		} else {
-			access = "rbind:rw"
+			opts = append(opts, "rw")
 		}
-		mount.Type = "bind"
-		mount.Source = path.Join("/var", disks[i].FileLocation)
-		if disks[i].MountDir != "" {
-			mount.Destination = disks[i].MountDir
-		} else {
-			mount.Destination = path.Join("/mnt", disks[i].DisplayName)
+
+		dests := []string{fmt.Sprintf("/dev/eve/volumes/by-id/%d", id)}
+		if disk.DisplayName != "" {
+			dests = append(dests, "/dev/eve/volumes/by-name/"+disk.DisplayName)
+		}
+		if disk.MountDir != "" {
+			if path.Join(disk.MountDir) != "/" {
+				dests = append(dests, disk.MountDir)
+			} else if path.Join(root, disk.MountDir) != "/" {
+				dests = append(dests, "..")
+			}
```

Comment:
Do we need this? In which case it will get trigger?

Reply:
Hey @zed-rishabh -- I just rewrote this to make it more streamlined -- please check it out